frontend/src/metabase/selectors/app.ts

```diff
@@ -148,3 +149,22 @@ export const getIsQuestionLineageVisible = createSelector(
     !originalQuestion.isDataset() &&
     PATHS_WITH_QUESTION_LINEAGE.some(pattern => pattern.test(path)),
 );
+
+export const getSettings = createSelector(
+  (state: State) => state.settings,
+  settings => settings.values,
+);
+
+export const getBannerMessageDescriptor = createSelector(
```

Comment:
Could we make this `getTokenStatus` and move checks for displaying the banner to the component itself? In that case it would accept `isAdmin` and `tokenStatus` props